engine/gamesys/src/gamesys/scripts/script_resource.cpp

```diff
@@ -474,6 +474,36 @@ static int SetTexture(lua_State* L)
     return 0;
 }
 
+/*# Update internal sound resource
+ * Update internal sound resource (wavc/oggc) with new data
+ *
+ * @name resource.set_sound
+ *
+ * @param path [type:hash|string] The path to the resource
+ * @param buffer [type:string] A lua string containing the binary sound data
+ */
+static int SetSound(lua_State* L) {
+    DM_LUA_STACK_CHECK(L, 0);
+
+    // get resource path as hash
+    const dmhash_t path_hash = dmScript::CheckHashOrString(L, 1);
+    // get the sound buffer
+    luaL_checktype(L, 2, LUA_TSTRING);
```

Comment:
👍 
I think we can add support for buffers later (if we need it).
Or what do you think @britzl ?

Reply:
I agree! Let's start with strings!